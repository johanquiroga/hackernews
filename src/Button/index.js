import React from 'react';
import PropTypes from 'prop-types';
import { withLoading } from '../HOC';

const Button = ({ onClick, className, children }) =>
  <button
    onClick={onClick}
    className={className}
    type="button"
  >
    {children}
  </button>

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default Button;

export const ButtonWithLoading = withLoading(Button);
