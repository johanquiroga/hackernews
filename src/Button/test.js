import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { compose } from 'recompose';

import Button from './index.js';
import Loading from '../Loading';
import { withLoading } from '../HOC';

import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { faSpinner, faSort, faSortUp, faSortDown } from '@fortawesome/fontawesome-free-solid';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  const props = {
    className: 'button',
    onClick() {console.log('Button clicked')}
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button { ...props }>Give me More</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('sets the given class', () => {
    const element = render(<Button { ...props }>Give me More</Button>);
    expect(element.is('.button')).toBe(true);
  });

  it('sets the correct child', () => {
    const element = render(<Button { ...props }>Give me More</Button>);
    expect(element.text()).toEqual('Give me More');
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Button { ...props }>Give me More</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ButtonWithLoading', () => {
  const props = {
    onClick() {console.log('Button clicked')}
  };

  const ButtonWithLoading = compose(
    withLoading
  )(Button);

  it('renders button without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ButtonWithLoading isLoading={false} {...props}>More</ButtonWithLoading>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders Loading without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ButtonWithLoading isLoading={true} {...props}>More</ButtonWithLoading>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('mounts a Button Component', () => {
    const component = mount(<ButtonWithLoading isLoading={false} {...props}>More</ButtonWithLoading>);
    expect(component.find(Button).length).toBe(1);
  });

  it('mounts a Loading Component', () => {
    const component = mount(<ButtonWithLoading isLoading={true} {...props}>More</ButtonWithLoading>);
    expect(component.find(Loading).length).toBe(1);
  });

  test('has a valid button snapshot', () => {
    const component = renderer.create(<ButtonWithLoading isLoading={false} {...props}>More</ButtonWithLoading>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('has a valid loading snapshot', () => {
    const component = renderer.create(<ButtonWithLoading isLoading={true} {...props}>More</ButtonWithLoading>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
