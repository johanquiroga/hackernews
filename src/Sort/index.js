import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';

import Button from '../Button';

const Sort = ({sortKey, isSortReverse, activeSortKey, onSort, children}) => {
  const sortClass = classNames(
    'button-inline',
    { 'button-active': sortKey === activeSortKey }
  );

  const sortIcon = sortKey === activeSortKey
    ? isSortReverse
      ? 'sort-up'
      : 'sort-down'
    : 'sort';

  return (
    <Button
      onClick={() => onSort(sortKey)}
      className={sortClass}
    >
      {children} <FontAwesomeIcon icon={sortIcon} size="xs" />
    </Button>
  );
}

Sort.propTypes = {
  sortKey: PropTypes.string.isRequired,
  isSortReverse: PropTypes.bool.isRequired,
  activeSortKey: PropTypes.string.isRequired,
  onSort: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default Sort;
