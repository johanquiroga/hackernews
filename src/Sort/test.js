import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Sort from './index.js';
import Button from '../Button';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { faSpinner, faSort, faSortUp, faSortDown } from '@fortawesome/fontawesome-free-solid';

Enzyme.configure({ adapter: new Adapter() });

describe('Sort', () => {
  const props = {
    sortKey: 'NONE',
    isSortReverse: false,
    activeSortKey: 'TITLE',
    onSort() {console.log(sorting)}
  };

  it('renders Sort without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Sort {...props}>Sort</Sort>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('mounts a Sort Component', () => {
    const component = mount(<Sort {...props}>Sort</Sort>);
    expect(component.find(Sort).length).toBe(1);
  });

  it('shows sort icon', () => {
    const wrapper = shallow(<Sort {...props}>Sort</Sort>);
    expect(wrapper.find(Button).find(FontAwesomeIcon).prop('icon')).toBe('sort');
  });

  it('shows sort-down icon', () => {
    const wrapper = mount(<Sort {...props}>Sort</Sort>);
    wrapper.setProps({ activeSortKey: 'NONE' });
    expect(wrapper.find(Button).find(FontAwesomeIcon).prop('icon')).toBe('sort-down');
  });

  it('shows sort-up icon', () => {
    const wrapper = mount(<Sort {...props}>Sort</Sort>);
    wrapper.setProps({ activeSortKey: 'NONE', isSortReverse: true });
    expect(wrapper.find(Button).find(FontAwesomeIcon).prop('icon')).toBe('sort-up');
  });

  test('has a valid Sort snapshot', () => {
    const component = renderer.create(<Sort {...props}>Sort</Sort>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
