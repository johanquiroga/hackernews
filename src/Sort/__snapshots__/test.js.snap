// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort has a valid Sort snapshot 1`] = `
<button
  className="button-inline"
  onClick={[Function]}
  type="button"
>
  Sort
   
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-sort fa-w-10 fa-xs "
    data-icon="sort"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
